lib/src/deprecation.dart

```diff
@@ -68,16 +66,23 @@ enum Deprecation {
   /// The Dart Sass version this feature was first deprecated in.
   ///
   /// For deprecations that have existed in all versions of Dart Sass, this
-  /// should be 0.0.0. For deprecations that are not yet active, this should be
-  /// null.
+  /// should be 0.0.0. For deprecations not related to a specific Sass version,
+  /// this should be null.
   final String? deprecatedIn;
 
   /// A description of this deprecation that will be displayed in the CLI usage.
   ///
   /// If this is null, the given deprecation will not be listed.
   final String? description;
 
-  const Deprecation(this.id, {required this.deprecatedIn, this.description});
+  /// Whether this deprecation will occur in the future.
+  ///
+  /// If this is true, `deprecatedIn` should be null, since we do not yet know
+  /// what version of Dart Sass this deprecation will be live in.
```

Comment:
If you make this a real class, you can also verify this in a constructor `assert()`.

Reply:
Accomplished this in the enum form by having separate constructors for regular and future deprecations.